Source/Particles/Collision/BackgroundMCCCollision.cpp

```diff
@@ -0,0 +1,376 @@
+    // query for a list of collision processes
+    // these could be elastic, excitation, charge_exchange, back, etc.
+    amrex::Vector<std::string> scattering_process_names;
+    pp.queryarr("scattering_processes", scattering_process_names);
+
+    // create a vector of MCCProcess objects from each scattering
+    // process name
+    for (auto scattering_process : scattering_process_names) {
+        std::string kw_cross_section = scattering_process + "_cross_section";
+        std::string cross_section_file;
+        pp.query(kw_cross_section.c_str(), cross_section_file);
+
+        amrex::Real energy = 0.0;
+        // if the scattering process is excitation or ionization get the
+        // energy associated with that process
+        if (scattering_process.find("excitation") != std::string::npos ||
+            scattering_process.find("ionization") != std::string::npos) {
+            std::string kw_energy = scattering_process + "_energy";
+            pp.get(kw_energy.c_str(), energy);
+        }
+
+        auto process = new MCCProcess(scattering_process, cross_section_file, energy);
+
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(process->m_type != MCCProcessType::INVALID,
+                                         "Cannot add an unknown MCC process type");
+
+        // if the scattering process is ionization get the secondary species
+        // only one ionization process is supported, the vector
+        // m_ionization_processes is only used to make it simple to calculate
+        // the maximum collision frequency with the same function used for
+        // particle conserving processes
+        if (process->m_type == MCCProcessType::IONIZATION) {
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(!ionization_flag,
+                                             "Background MCC only supports a single ionization process");
+            ionization_flag = true;
+
+            std::string secondary_species;
+            pp.get("ionization_species", secondary_species);
+            m_species_names.push_back(secondary_species);
+
+            m_ionization_processes.push_back(process);
+        } else {
+            m_scattering_processes.push_back(process);
+        }
+
+        amrex::Gpu::synchronize();
+    }
+}
+
+/** Calculate the maximum collision frequency using a fixed energy grid that
+ *  ranges from 1e-4 to 5000 eV in 0.2 eV increments
+ */
+amrex::Real
+BackgroundMCCCollision::get_nu_max(amrex::Gpu::ManagedVector<MCCProcess*> const& mcc_processes)
+{
+    amrex::Real nu, nu_max = 0.0;
+
+    for (double E = 1e-4; E < 5000; E+=0.2) {
+        amrex::Real sigma_E = 0.0;
+
+        // loop through all collision pathways
+        for (const auto &scattering_process : mcc_processes) {
+            // get collision cross-section
+            sigma_E += scattering_process->getCrossSection(E);
+        }
+
+        // calculate collision frequency
+        nu = (
+              m_background_density * std::sqrt(2.0 / m_mass1 * PhysConst::q_e)
```

Comment:
avoids sqrt promotion to double in SP compilation:
```suggestion
              m_background_density * std::sqrt(2.0_rt / m_mass1 * PhysConst::q_e)
```